protocols/gossipsub/src/protocol.rs

```diff
@@ -319,6 +400,76 @@ impl Decoder for GossipsubCodec {
     }
 }
 
+/// Signs a gossipsub message, adding the key and signature fields.
+fn sign_message(keypair: &Keypair, message: &mut rpc_proto::Message) -> Result<(), io::Error> {
+    let mut buf = Vec::with_capacity(message.encoded_len());
+    message
+        .encode(&mut buf)
+        .expect("Buffer has sufficient capacity");
+    // the signature is over the bytes "libp2p-pubsub:<protobuf-message>"
+    let mut signature_bytes = b"libp2p-pubsub:".to_vec();
```

Comment:
Out of curiosity, what is the purpose of prefixing the signature with `libp2p-pubsub`? I can't find an explanation in the [spec](https://github.com/libp2p/specs/tree/master/pubsub#message-signing).

Reply:
Neither could I. 
Just speculating, but it ensures this signature can only be used in this protocol I guess. If some other protocol where using the same protobuf format, the signature would be invalid unless those protocols also required this string prefix.

Reply:
The go implementations does add this prefix: https://github.com/libp2p/go-libp2p-pubsub/blob/49274b0e8aecdf6cad59d768e5702ff00aa48488/sign.go#L12

Reply:
Yep, it's in the spec, so they wouldn't be following the spec if they didn't.

Reply:
Ah sorry, I misread the initial comment. Don't mind me :grimacing: